tests/components/amberelectric/conftest.py

```diff
@@ -13,3 +59,103 @@ def mock_setup_entry() -> Generator[AsyncMock]:
         "homeassistant.components.amberelectric.async_setup_entry", return_value=True
     ) as mock_setup_entry:
         yield mock_setup_entry
+
+
+@pytest.fixture
+async def setup_general(hass: HomeAssistant) -> AsyncGenerator[Mock]:
+    """Set up general channel."""
+    MockConfigEntry(
+        domain="amberelectric",
+        data={
+            CONF_SITE_NAME: "mock_title",
+            CONF_API_TOKEN: MOCK_API_TOKEN,
+            CONF_SITE_ID: GENERAL_ONLY_SITE_ID,
+        },
+    ).add_to_hass(hass)
+
+    instance = Mock()
+    with patch(
+        "amberelectric.AmberApi",
+        return_value=instance,
+    ) as mock_update:
+        instance.get_current_prices = Mock(return_value=GENERAL_CHANNEL)
+        assert await async_setup_component(hass, DOMAIN, {})
+        await hass.async_block_till_done()
+        yield mock_update.return_value
+
+
+@pytest.fixture
+async def setup_general_and_controlled_load(
+    hass: HomeAssistant,
+) -> AsyncGenerator[Mock]:
+    """Set up general channel and controller load channel."""
+    MockConfigEntry(
+        domain="amberelectric",
+        data={
+            CONF_API_TOKEN: MOCK_API_TOKEN,
+            CONF_SITE_ID: GENERAL_AND_CONTROLLED_SITE_ID,
+        },
+    ).add_to_hass(hass)
+
+    instance = Mock()
+    with patch(
+        "amberelectric.AmberApi",
+        return_value=instance,
+    ) as mock_update:
+        instance.get_current_prices = Mock(
+            return_value=GENERAL_CHANNEL + CONTROLLED_LOAD_CHANNEL
+        )
```

Comment:
would it maybe make sense to create a fixture for this? Maybe the `bosch_alarm` tests can give you some inspiration. this way we don't have to duplicate this code